providers/azure/aci.go

```diff
@@ -329,6 +332,69 @@ func (p *ACIProvider) GetContainerLogs(namespace, podName, containerName string,
 	return logContent, err
 }
 
+// ExecInContainer executes a command in a container in the pod, copying data
+// between in/out/err and the container's stdin/stdout/stderr.
+func (p *ACIProvider) ExecInContainer(name string, uid types.UID, container string, cmd []string, in io.Reader, out, errstream io.WriteCloser, tty bool, resize <-chan remotecommand.TerminalSize, timeout time.Duration) error {
+	cg, _, _ := p.aciClient.GetContainerGroup(p.resourceGroup, name)
+
+	// Set default terminal size
+	terminalSize := remotecommand.TerminalSize{
+		Height: 60,
+		Width:  120,
+	}
+
+	if resize != nil {
+		terminalSize = <-resize // Receive terminal resize event if resize stream is present
+	}
+
+	xcrsp, _ := p.aciClient.LaunchExec(p.resourceGroup, cg.Name, container, cmd[0], terminalSize)
+
+	wsUri := xcrsp.WebSocketUri
+	password := xcrsp.Password
+
+	c, _, _ := websocket.DefaultDialer.Dial(wsUri, nil)
+	c.WriteMessage(websocket.TextMessage, []byte(password)) // Websocket password needs to be sent before WS terminal is active, is this ACI specific?
+
+	// Cleanup on exit
+	defer c.Close()
+	defer out.Close()
+
+	if in != nil {
+		// Inputstream
+		go func() {
+			for {
+				var msg = make([]byte, 512)
+				_, err := in.Read(msg)
```

Comment:
I tried the image, the exec works well, except one minor issue: it doesn't respond to "tab". Do you have any idea?

Reply:
Whoops you found a bug. I should check how many bytes were actually read from stdin. I was sending 512 bytes every time, mostly zeroes. Apparently that breaks tab functionality.

Fixed it, will push.